torch/_dynamo/variables/tensor.py

```diff
@@ -333,6 +333,23 @@ def method_attr__version(self, tx):
             tx, [self], {}
         )
 
+    def call_hasattr(self, tx, name):
+        try:
+            self.var_getattr(tx, name)
+            ret_val = True
+        except (NotImplementedError, AttributeError):
```

Comment:
Are you sure about this? `var_getattr` is Dynamo codebase, are we supposed to return AttributeError. If there is an overridden `__getattr__` function, we might return `ObservedExpection`

Reply:
https://github.com/pytorch/pytorch/blob/82242a258a7728289caf64dfaabef552a2c994b8/torch/_dynamo/variables/tensor.py#L260

Reply:
If the getattr is defined we return `NotImplementedError`